plugin/core/typing.py

```diff
@@ -21,9 +22,11 @@
     from typing import Set
     from typing import Tuple
     from typing import Type
-    from typing import TypedDict
     from typing import TypeVar
     from typing import Union
+    from typing_extensions import NotRequired
+    from typing_extensions import Required
+    from typing_extensions import TypedDict
```

Comment:
typing_extensions is a third-party package, so this won't work on python 3.8.

Reply:
In 3.8 we could keep using real imports for type checking and dummy ones for the plugin host, like we do now.
Then:
 - this won't really matter
 - if we set python version to 3.11 for pyright then we can import those from `typing`

One thing I'm not sure about (although it doesn't matter for ST) is how both pyright and mypy don't have problems with this `typing_extensions` import. I guess that they either come with own copies or use system ones?

Reply:
Addressing in https://github.com/sublimelsp/LSP/pull/2067